src/main/java/cloud/fogbow/common/plugins/cloudidp/googlecloud/GoogleCloudIdentityProviderPlugin.java

```diff
@@ -0,0 +1,81 @@
+package cloud.fogbow.common.plugins.cloudidp.googlecloud;
+
+import java.security.PrivateKey;
+import java.security.interfaces.RSAPrivateKey;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+
+import cloud.fogbow.common.constants.GoogleCloudConstants;
+import cloud.fogbow.common.constants.HttpMethod;
+import cloud.fogbow.common.exceptions.UnauthenticatedUserException;
+import cloud.fogbow.common.models.GoogleCloudUser;
+import cloud.fogbow.common.plugins.cloudidp.CloudIdentityProviderPlugin;
+import cloud.fogbow.common.util.CryptoUtil;
+import cloud.fogbow.common.util.connectivity.HttpRequestClient;
+import cloud.fogbow.common.util.connectivity.HttpResponse;
+
+public class GoogleCloudIdentityProviderPlugin implements CloudIdentityProviderPlugin<GoogleCloudUser> {
+
+	private final String endPoint;
+	private final String TOKEN_ENDPOINT = "/token";
+	private final static String URL = "https://oauth2.googleapis.com";
+	private final String SCOPE_VALUE = "https://www.googleapis.com/auth/compute";
+	private final String GRANT_TYPE_VALUE = "urn:ietf:params:oauth:grant-type:jwt-bearer";
+	private final String BEGIN_PRIVATE_KEY = "-----BEGIN PRIVATE KEY-----";
+	private final String END_PRIVATE_KEY = "-----END PRIVATE KEY-----";
+
+	public GoogleCloudIdentityProviderPlugin(String identityUrl) {
+		this.endPoint = identityUrl + TOKEN_ENDPOINT;
+	}
+
+	public GoogleCloudIdentityProviderPlugin() {
+		this(URL);
+	}
+
+	@Override
+	public GoogleCloudUser getCloudUser(Map<String, String> userCredentials) throws UnauthenticatedUserException {
+
+		try {
+
+			String email = userCredentials.get(GoogleCloudConstants.Identity.EMAIL);
+			String privateKey = userCredentials.get(GoogleCloudConstants.Identity.PRIVATE_KEY);
+
+			privateKey = privateKey.replace(BEGIN_PRIVATE_KEY, "").replace(END_PRIVATE_KEY, "");
+
+			PrivateKey objKey = CryptoUtil.getPrivateKeyFromString(privateKey);
+
+			String signedJwt = this.createJwt(objKey, email);
+			Map<String, String> requestBody = new HashMap<String, String>();
+			requestBody.put(GoogleCloudConstants.Identity.GRANT_TYPE_KEY, GRANT_TYPE_VALUE);
+			requestBody.put(GoogleCloudConstants.Identity.ASSERTION_KEY, signedJwt);
+
+			Map<String, String> header = new HashMap<String, String>();
+			header.put("Content-Type", "application/json");
+
+			HttpResponse response = HttpRequestClient.doGenericRequest(HttpMethod.POST, endPoint, header, requestBody);
+			CreateAuthenticationResponse responseObject = CreateAuthenticationResponse.fromJson(response.getContent());
+
+			return new GoogleCloudUser(email, email, responseObject.getToken(),
+					userCredentials.get(GoogleCloudConstants.Identity.PROJECT_ID_KEY));
+
+		} catch (Exception e) {
+			throw new UnauthenticatedUserException();
```

Comment:
Seria interessante manter a mensagem de erro da exceção para facilitar na hora de depurar o código.

```suggestion
			throw new UnauthenticatedUserException(e.getMessage());
```

Reply:
Mensagem atribuída a exceção.